sdk/messaging/azservicebus/admin/admin_client_rules.go

```diff
@@ -0,0 +1,558 @@
+	// Subject will be matched against the Subject property for the message.
+	Subject *string
+
+	// To will be matched against the To property for the message.
+	To *string
+}
+
+// RuleProperties are the properties for a rule.
+type RuleProperties struct {
+	// Name is the name of this rule.
+	Name string
+
+	// Filter is the filter for this rule.
+	Filter *Filters
+
+	// Action is the action for this rule.
+	Action *Actions
+}
+
+// CreateRuleResponse contains the response fields for Client.CreateRule
+type CreateRuleResponse struct {
+	RuleProperties
+}
+
+// CreateRuleOptions contains the optional parameters for Client.CreateRule
+type CreateRuleOptions struct {
+	// Name is the name of the rule or nil, which will default to $Default
+	Name *string
+
+	// Filter is the filter for this rule
+	Filter *Filters
+
+	// Action is the action for this rule
+	Action *Actions
+}
+
+// CreateRule creates a rule that can filter and update message for a subscription.
+func (ac *Client) CreateRule(ctx context.Context, topicName string, subscriptionName string, options *CreateRuleOptions) (CreateRuleResponse, error) {
+	ruleName := ""
+
+	if options != nil && options.Name != nil {
+		ruleName = *options.Name
+	}
+
+	resp, _, err := ac.createOrUpdateRule(ctx, topicName, subscriptionName, RuleProperties{
+		Name:   ruleName,
+		Filter: options.Filter,
+		Action: options.Action,
+	}, true)
+
+	if err != nil {
+		return CreateRuleResponse{}, err
+	}
+
+	return CreateRuleResponse{RuleProperties: *resp}, nil
+}
+
+// GetRuleResponse contains the response fields for Client.GetRule
+type GetRuleResponse struct {
+	// RuleProperties for the rule.
+	RuleProperties
+}
+
+// GetRuleOptions contains the optional parameters for Client.GetRule
+type GetRuleOptions struct {
+	// For future expansion
+}
+
+// GetRule gets a rule for a subscription.
+func (ac *Client) GetRule(ctx context.Context, topicName string, subscriptionName string, ruleName string, options *GetRuleOptions) (*GetRuleResponse, error) {
+	var ruleEnv *atom.RuleEnvelope
+
+	_, err := ac.em.Get(ctx, fmt.Sprintf("/%s/Subscriptions/%s/Rules/%s", topicName, subscriptionName, ruleName), &ruleEnv)
+
+	if err != nil {
+		return mapATOMError[GetRuleResponse](err)
+	}
+
+	props, err := newRuleProperties(ruleEnv)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &GetRuleResponse{
+		RuleProperties: *props,
+	}, nil
+}
+
+// ListRulesResponse contains the response fields for the pager returned from Client.ListRules.
+type ListRulesResponse struct {
+	// Rules are all the rules for the page.
+	Rules []RuleProperties
+}
+
+// ListRulesOptions contains the optional parameters for Client.ListRules
+type ListRulesOptions struct {
+	// MaxPageSize is the maximum size of each page of results.
+	MaxPageSize int32
+}
+
+// ListRules creates a pager that can list rules for a subscription.
+func (ac *Client) ListRules(topicName string, subscriptionName string, options *ListRulesOptions) *runtime.Pager[ListRulesResponse] {
```

Comment:
This should be named `NewListRulesPager(...)`

Reply:
Will rename (all others too)

Reply:
Renamed